internal/tools/securitypolicy/main.go

```diff
@@ -172,8 +184,61 @@ func createPolicyFromConfig(config Config) (sp.SecurityPolicy, error) {
 			container.Layers = append(container.Layers, hashString)
 		}
 
+		// add rules for all known environment variables from the configuration
+		// these are in addition to "other rules" from the policy definition file
+		config, err := img.ConfigFile()
+		if err != nil {
+			return p, err
+		}
+		for _, env := range config.Config.Env {
+			rule := sp.SecurityPolicyEnvironmentVariableRule{
+				Strategy: "string",
+				Rule:     env,
+			}
+
+			container.EnvRules = append(container.EnvRules, rule)
+		}
+
+		// cri adds TERM=xterm for all workload containers. we add to all containers
+		// to prevent any possble erroring
+		rule := sp.SecurityPolicyEnvironmentVariableRule{
+			Strategy: "string",
+			Rule:     "TERM=xterm",
+		}
+
+		container.EnvRules = append(container.EnvRules, rule)
+
 		p.Containers = append(p.Containers, container)
 	}
 
 	return p, nil
 }
+
+func validateEnvRules(rules []EnvironmentVariableRule) error {
+	for _, rule := range rules {
+		switch rule.Strategy {
+		case "re2":
+			_, err := regexp.Compile(rule.Rule)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func convertEnvironmentVariableRules(toml []EnvironmentVariableRule) []sp.SecurityPolicyEnvironmentVariableRule {
+	json := make([]sp.SecurityPolicyEnvironmentVariableRule, len(toml))
```

Comment:
Generally try and avoid naming variables as the same name as a package that's imported

Reply:
There's an item to address in a later PR in general. So this will be changing in the second half of September.